core/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -17,53 +17,297 @@
+      // when we create the Local Relation, example
+      // SELECT MAX(Column2) FROM (SELECT Column1 AS Column2 FROM TableName)
+      // the AggregateExpression contains a reference to Column2, instead of Column1
+      PhysicalOperation(fields, Nil, DeltaTable(i: TahoeLogFileIndex)))
+        if i.partitionFilters.isEmpty
+          && fields.forall {
+          case _: AttributeReference => true
+          case _ => false
+        }
+          && aggExpr.forall {
+          case Alias(AggregateExpression(
+            Count(Seq(Literal(1, _))) | Min(_) | Max(_), Complete, false, None, _), _) => true
+          case _ => false
+        } =>
+        Some(i)
+      // When all columns are selected, there are no Project/PhysicalOperation
       case Aggregate(
-          Nil,
-          Seq(Alias(AggregateExpression(Count(Seq(Literal(1, _))), Complete, false, None, _), _)),
-          PhysicalOperation(_, Nil, DeltaTable(i: TahoeLogFileIndex))) if i.partitionFilters.isEmpty
-        => extractGlobalCount(i)
+      Nil,
+      aggExpr: Seq[Alias],
+      DeltaTable(i: TahoeLogFileIndex))
+        if i.partitionFilters.isEmpty
+          && aggExpr.forall {
+          case Alias(AggregateExpression(
+            Count(Seq(Literal(1, _))) | Min(_) | Max(_), Complete, false, None, _), _) => true
+          case _ => false
+        } =>
+        Some(i)
       case _ => None
     }
+  }
+
+  /** Return the number of rows in the table or `None` if we cannot calculate it from stats */
+  private def extractGlobalCount(tahoeLogFileIndex: TahoeLogFileIndex): Option[Long] = {
+    // account for deleted rows according to deletion vectors
+    val dvCardinality = coalesce(col("deletionVector.cardinality"), lit(0))
+    val numLogicalRecords = (col("stats.numRecords") - dvCardinality).as("numLogicalRecords")
+    val row = getDeltaScanGenerator(tahoeLogFileIndex).filesWithStatsForScan(Nil)
+      .agg(
+        sum(numLogicalRecords),
+        // Calculate the number of files missing `numRecords`
+        count(when(col("stats.numRecords").isNull, 1)))
+      .first
+
+    // The count agg is never null. A non-zero value means we have incomplete stats; otherwise,
+    // the sum agg is either null (for an empty table) or gives an accurate record count.
+    if (row.getLong(1) > 0) return None
+    val numRecords = if (row.isNullAt(0)) 0 else row.getLong(0)
+    Some(numRecords)
+  }
+
+  val columnStatsSupportedDataTypes: HashSet[DataType] = HashSet(
+    ByteType,
+    ShortType,
+    IntegerType,
+    LongType,
+    FloatType,
+    DoubleType,
+    DateType)
+
+  case class DeltaColumnStat(
+    min: Any,
+    max: Any,
+    nullCount: Option[Long],
+    distinctCount: Option[Long])
+
+  def extractGlobalColumnStats(tahoeLogFileIndex: TahoeLogFileIndex):
+  CaseInsensitiveMap[DeltaColumnStat] = {
+
+    // TODO Update this to work with DV (https://github.com/delta-io/delta/issues/1485)
```

Comment:
Should we handle the DV support? The read support is already there. Check out the `DeletionVectorsSuite`

Reply:
I need to think more about how it will work. I added distinctCount (for partition values only) and nullCount just for completeness, it is probably better to remove them.

For min and max, I need to check if the value found are from a AddFile.stats where tightBounds=true.

Reply:
It seems the implementation to update the stats tightBounds is still not done. We need it to create the unit tests.

Reply:
Should we wait for it? Or could we proceed and address DV later?

Reply:
We can add support later, but at least we shouldn't proceed with the optimization if DVs are preset. Otherwise it will give wrong results.

Reply:
Done.
We have ETA when DV will be released?